File: Semestrul 5/PPL/Laborator1/src/main/java/ro/ppd2024/util/SequentialConvolution.java
package ro.ppd2024.util;

public class SequentialConvolution {
    public static void convolute(int[][] F, int n, int m, int[][] V, int[][] C, int k) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                V[i][j] = Convolution.convolute(F, n, m, i, j, C, k);
            }
        }
    }
}

File: Semestrul 5/PPD/Laborator1/ConvolutionCpp/util/SequentialConvolution.cpp
#include "SequentialConvolution.h"
#include "Convolution.h"

template <typename TMatrix, typename TFilter>
void SequentialConvolution::performTask(TMatrix F, int n, int m, TMatrix V, TFilter C, int k) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            V[i][j] = convolute(F, n, m, i, j, C, k);
        }
    }
}
